cmd/start.go

```diff
@@ -79,6 +80,17 @@ func start(newProxy ProxyFactory) error {
 		logger.Errorf("failed to load in-cluster config: %v", err)
 	}
 
+	metricsPort := viper.GetString("metricsPort")
+	go func() {
+		err := metrics.Start(metrics.Config{
+			Port:   metricsPort,
+			Logger: zap.S(),
+		})
+		if err != nil {
+			logger.Errorf("failed to start metrics server: %v", err)
+		}
```

Comment:
- If the metrics server fails to start (e.g. port is not available to bind), this would silently fail. We should call `logger.Fatal` here so the program should stop immediately. The better way is to use [`errgroup`](https://pkg.go.dev/golang.org/x/sync/errgroup) to setup both app and metrics server (and also support graceful shutdown) but we'll do another PR for that.
-  We should use the `zap.Error()` like [this](https://github.com/Twingate/kubernetes-access-gateway/blob/14873853c4cc97ecc24e96699d82899b237931b6/internal/httpproxy/http_proxy.go#L195) instead of embedding the error in the log message.

Reply:
Yes, I think metrics collection is as critical. It's better to stop the gateway than running without Telemetry.